package.json

```diff
@@ -124,13 +124,16 @@
     "documentation": "5.3.2",
     "eslint": "4.9.0",
     "eslint-plugin-jsdoc": "3.1.3",
+    "eslint-config-prettier": "2.6.0",
+    "eslint-plugin-prettier": "2.3.1",
```

Comment:
What are these two deps for? Also any reason we aren't floating on these?

Reply:
Sorry, it's my bad, my global `.npmrc` dropped id.

`eslint-config-prettier` and `eslint-plugin-prettier` integrate ESLint with `prettier`, so ESLint can detect rules against prettier and IDEs can recognize it.

Read more about it: https://prettier.io/docs/en/usage.html#eslint

Reply:
Okay, I just realized that with this I killed the separate linting and style-checking, this PR needs a little bit more work.

Reply:
I fixed it, the new version uses `ESLint` with the environment variable trick to separate linting and style checking. The `make check-style` uses ESLint with `prettier` rules. The `make fix-style` uses `prettier` to re-print the code.